### PR TITLE
Consume generator as a package

### PR DESCRIPTION
**Pull request by pakrym**
Generator, Autorest shared soure and targets are all packed into a single nuget package now.

### CODE REVIEW

File: eng/CodeGeneration.targets [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
```
Comment (pakrym) [inline eng/CodeGeneration.targets:3]: This target file goes away when the last user of `TemporaryUsePreviousGeneratorVersion` is gone.

File: eng/CodeGeneration.targets [hunk 2]
```diff
@@ -38,17 +33,9 @@
     <Error Text="Following GitHub URLs do not contain commit hash: @(GithubUrlsWithoutHash) please use permalinks for code generation inputs (see https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files) " Condition="'@(GithubUrlsWithoutHash)' != ''" />
 
     <RemoveDir Directories="$(MSBuildProjectDirectory)/Generated"/>
-    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) $(AutoRestInput) $(AutoRestAdditionalParameters) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(_SharedCodeDirectory)" />
+    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) $(AutoRestInput) $(AutoRestAdditionalParameters) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(AutoRestSharedCodeDirectory).." />
```
Comment (pakrym) [inline eng/CodeGeneration.targets:36]: Reuse shared sources from the package.

**Review by heaths — APPROVED** [review]
Pretty slick.

File: eng/Directory.Build.Data.props [hunk 3]
```diff
@@ -152,4 +152,12 @@
     <!-- Disable doc comments for test projects -->
     <DocumentationFile></DocumentationFile>
   </PropertyGroup>
+
+  <!-- Code generation settings -->
+  <PropertyGroup>
+    <!-- Don't use code generator bundled Azure.Core shared source for code generation -->
+    <AzureCoreSharedCodeDirectory>$(AzureCoreSharedSources)</AzureCoreSharedCodeDirectory>
```
Comment (weshaggard) [inline eng/Directory.Build.Data.props:159]: I'm not sure I understand this? If the shared sources in the nuget package match the generator why wouldn't we always use those?

Comment (pakrym) [reply eng/Directory.Build.Data.props:159]: Generator nuget package carries 2 kinds of shared sources:

1. Generator specific sources
2. Parts of Azure.Core shared that we sync from azure-sdk-for-net to autorest.csharp and are required for generated libraries.

We always use the former from the package but considering this repo has the original versions of Azure.Core shared files we doing used them from the package.

**Pull request merged**